src/main/kotlin/no/digdir/catalog_admin_service/controller/CodeListController.kt

```diff
@@ -40,4 +53,34 @@ open class CodeListController(private val codeListService: CodeListService, priv
             ResponseEntity(HttpStatus.FORBIDDEN)
         }
 
+    @DeleteMapping(value = ["/{codeListId}"])
+    fun deleteCodeList(
+        @AuthenticationPrincipal jwt: Jwt,
+        @PathVariable catalogId: String,
+        @PathVariable codeListId: String
+    ): ResponseEntity<Unit> =
+        if (endpointPermissions.hasOrgAdminPermission(jwt, catalogId) && codeListService.getCodeListById(catalogId, codeListId) != null) {
+            codeListService.deleteCodeListById(codeListId)
+            ResponseEntity(HttpStatus.NO_CONTENT)
+        } else ResponseEntity(HttpStatus.FORBIDDEN)
```

Comment:
Tror det blir mer korrekt med 404 om noen spør på feil katalog
```suggestion
        when {
            !endpointPermissions.hasOrgAdminPermission(jwt, catalogId) -> ResponseEntity(HttpStatus.FORBIDDEN)
            codeListService.getCodeListById(catalogId, codeListId) == null -> ResponseEntity(HttpStatus.NOT_FOUND)
            else -> {
                codeListService.deleteCodeListById(codeListId)
                ResponseEntity(HttpStatus.NO_CONTENT)
            }
        }
```